src/HttpClientFactory/Http/src/Logging/HttpHeadersLogValue.cs

```diff
@@ -94,15 +96,23 @@ public override string ToString()
                     builder.Append(kvp.Key);
                     builder.Append(": ");
 
-                    foreach (var value in (IEnumerable<object>)kvp.Value)
+                    if (_isSensitiveHeader(kvp.Key))
                     {
-                        builder.Append(value);
-                        builder.Append(", ");
+                        builder.Append("*");
+                        builder.AppendLine();
                     }
+                    else
+                    {
+                        foreach (var value in (IEnumerable<object>)kvp.Value)
+                        {
+                            builder.Append(value);
+                            builder.Append(", ");
+                        }
 
-                    // Remove the extra ', '
-                    builder.Remove(builder.Length - 2, 2);
-                    builder.AppendLine();
+                        // Remove the extra ', '
```

Comment:
Is it worth pulling this into an extension method on `StringBuilder` and calling `StringBuilder.AppendJoin` on `!NETSTANDARD2_0`?

Reply:
It might be useful to have a similar API on the `StringBuilder` as we already have on the `String` for joining elements.

Reply:
It doesn't have to be an extension method, but it could still be worth #ifdef'ing to call the "proper" `AppendJoin`, no? I don't really know how often this is called, but appending and then removing doesn't sound as optimized as [`AppendJoin`](https://github.com/dotnet/runtime/blob/4359ebfc9943608b34f411366b1e544ac45702b7/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L1297).

Reply:
that would be fine

Reply:
Could you please check whether I used `#if` correctly?

Reply:
Looks right. `#if NETCOREAPP ` is fine. You can remove the TODO